llarp/rpc/rpc_server.cpp

```diff
@@ -94,6 +94,210 @@ namespace llarp::rpc
     }
   }
 
+  void
+  HandleJSONCommand(
+      oxenmq::Message& msg,
+      std::string replyCmd,
+      std::function<void(nlohmann::json, ReplyFunction_t)> handleRequest)
+  {
+    const auto maybe = MaybeParseJSON(msg);
+    if (not maybe.has_value())
+    {
+      msg.send_back(replyCmd, CreateJSONError("failed to parse json"));
+      return;
+    }
+    if (not maybe->is_object())
+    {
+      msg.send_back(replyCmd, CreateJSONError("data not a json object"));
+      return;
+    }
+    try
+    {
+      handleRequest(*maybe, [defer = msg.send_later(), cmd = replyCmd](std::string result) {
+        defer(cmd, result);
+      });
+    }
+    catch (std::exception& ex)
+    {
+      msg.send_back(replyCmd, CreateJSONError(ex.what()));
+    }
+  }
+
+  void
+  GetExit(AbstractRouter* r, nlohmann::json obj, ReplyFunction_t reply)
+  {
+    if (r->IsServiceNode())
+    {
+      reply(CreateJSONError("not supported"));
+      return;
+    }
+    std::optional<llarp::service::Address> exit;
+    std::optional<std::string> lnsExit;
+    IPRange range;
+    bool map = true;
+    const auto exit_itr = obj.find("exit");
```

Comment:
this amount of find->if->set with json definitely warrants a helper function to simplify it.